rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -3231,4 +3252,91 @@ private boolean isXmlWorkflow(InputPart fileInputPart) {
+                    throws NotConnectedRestException, PermissionRestException {
+        return getTaskIdsByTag(sessionId, null, from, to, mytasks, running, pending, finished, offset, limit);
+    }
+    
+    @Override
+    public RestPage<String> getTaskIdsByTag(String sessionId, String taskTag, long from, long to, boolean mytasks,
+            boolean running, boolean pending, boolean finished, int offset, int limit)
+                    throws NotConnectedRestException, PermissionRestException {
+        Scheduler s = checkAccess(sessionId, "tasks");
+        PageBoundariesRest boundaries = getBoundaries(offset, limit);
+        Page<TaskId> page = null;
+        try {
+            page = s.getTaskIds(taskTag, from, to, mytasks, running, pending, finished,
+                    boundaries.getOffset(), boundaries.getLimit());
+            ArrayList<String> taskNames = new ArrayList<String>(page.getList().size());
+            for (TaskId taskId : page.getList()) {
+                taskNames.add(taskId.getReadableName());
+            }
+            return new RestPage<String>(taskNames, page.getSize());
+        } catch (NotConnectedException e) {
+            throw new NotConnectedRestException(e);
+        } catch (PermissionException e) {
+            throw new PermissionRestException(e);
+        }
+    }
+
+    @Override
+    public RestPage<TaskStateData> getTaskStates(String sessionId, long from, long to, boolean mytasks,
+            boolean running, boolean pending, boolean finished, int offset, int limit)
+                    throws NotConnectedRestException, PermissionRestException {
+        return getTaskStatesByTag(sessionId, null, from, to, mytasks, running, pending, finished, offset, limit);
+    }
+
+    @Override
+    public RestPage<TaskStateData> getTaskStatesByTag(String sessionId, String taskTag, long from, long to,
+            boolean mytasks, boolean running, boolean pending, boolean finished, int offset, int limit)
+                    throws NotConnectedRestException, PermissionRestException {
+        Scheduler s = checkAccess(sessionId, "tasks/tag/" + taskTag);
+        PageBoundariesRest boundaries = getBoundaries(offset, limit);
+        Page<TaskState> page = null;
+        try {
+            page = s.getTaskStates(taskTag, from, to, mytasks, running, pending, finished,
+                    boundaries.getOffset(), boundaries.getLimit());
+            List<TaskStateData> tasks = map(page.getList(), TaskStateData.class);
+            return new RestPage<TaskStateData>(tasks, page.getSize());
+        } catch (NotConnectedException e) {
+            throw new NotConnectedRestException(e);
+        } catch (PermissionException e) {
+            throw new PermissionRestException(e);
+        }
+    }
+    
+    private PageBoundariesRest getBoundaries(int offset, int limit) {
+        
+        if (offset < 0) offset = 0;
+        if (limit < 0 || offset > limit) limit = offset + TASKS_PAGE_SIZE;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00122)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "limit". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)